### PR TITLE
[Enhancement] Support consuming from Kafka with IPv6 in routine load task

### PR DESCRIPTION
**Pull request by qingzhongli**
## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: fe/fe-core/src/test/java/com/starrocks/analysis/CreateRoutineLoadStmtTest.java [hunk 1]
```diff
@@ -90,42 +90,84 @@ public static void beforeClass() throws Exception {
...
+                    + "(\n"
+                    + "\"desired_concurrent_number\"=\"3\",\n"
+                    + "\"max_batch_interval\" = \"20\",\n"
+                    + "\"max_filter_ratio\" = \"0.12\",\n"
+                    + "\"strict_mode\" = \"false\",\n"
+                    + "\"timezone\" = \"Asia/Shanghai\"\n"
+                    + ")\n"
+                    + "FROM KAFKA\n"
+                    + "(\n"
+                    + "\"kafka_broker_list\" = \"[kafkahost1]:9092,[kafkahost2]:9092\",\n"
```
Comment (rickif) [inline fe/fe-core/src/test/java/com/starrocks/analysis/CreateRoutineLoadStmtTest.java:148]: Why is the `kafkahost1` wrapped with `[]`?

File: fe/fe-core/src/test/java/com/starrocks/analysis/CreateRoutineLoadStmtTest.java [hunk 2]
```diff
@@ -333,288 +467,656 @@ public void testLoadColumns() {
...
+            Assert.assertEquals(createRoutineLoadStmt.getMaxErrorNum(), 10);
+            Assert.assertEquals(createRoutineLoadStmt.getEnclose(), 0);
+            Assert.assertEquals(createRoutineLoadStmt.getEscape(), 0);
+            Assert.assertEquals(createRoutineLoadStmt.isTrimspace(), false);
+        }
+
+        {
+            String createSQL = "CREATE ROUTINE LOAD db0.routine_load_1 ON t1 " +
+                    "PROPERTIES(\"max_error_number\" = \"10\") " +
+                    "FROM KAFKA(\"kafka_broker_list\" = \"xxxx:xxxx:xxxx:xxxx:xxxx:xxxx:xxxx:xxxx\",\"kafka_topic\" = \"topic_0\");";
```
Comment (rickif) [inline fe/fe-core/src/test/java/com/starrocks/analysis/CreateRoutineLoadStmtTest.java:840]: Should the ipv6 address be wrapped with `[]`?

File: fe/fe-core/src/test/java/com/starrocks/analysis/CreateRoutineLoadStmtTest.java (hunk 1 above)
Comment (qingzhongli) [reply fe/fe-core/src/test/java/com/starrocks/analysis/CreateRoutineLoadStmtTest.java:148]: It used to indicate that kafka broker is an IPv6 address

File: fe/fe-core/src/test/java/com/starrocks/analysis/CreateRoutineLoadStmtTest.java (hunk 2 above)
Comment (qingzhongli) [reply fe/fe-core/src/test/java/com/starrocks/analysis/CreateRoutineLoadStmtTest.java:840]: Thank you for your review, I will fix it

File: fe/fe-core/src/test/java/com/starrocks/analysis/CreateRoutineLoadStmtTest.java (hunk 1 above)
Comment (rickif) [reply fe/fe-core/src/test/java/com/starrocks/analysis/CreateRoutineLoadStmtTest.java:148]: I think the IPV6 address should be enclosed, but the host name shouldn't.

File: fe/fe-core/src/main/java/com/starrocks/sql/ast/CreateRoutineLoadStmt.java [hunk 3]
```diff
@@ -136,7 +136,7 @@ public class CreateRoutineLoadStmt extends DdlStmt {
     public static final String PULSAR_DEFAULT_INITIAL_POSITION = "pulsar_default_initial_position";
 
     private static final String NAME_TYPE = "ROUTINE LOAD NAME";
-    private static final String ENDPOINT_REGEX = "[-A-Za-z0-9+&@#/%?=~_|!:,.;]+[-A-Za-z0-9+&@#/%=~_|]";
+    private static final String ENDPOINT_REGEX = "[-A-Za-z0-9+&@#/%?=~_|!:,.;\\[\\]]+[-A-Za-z0-9+&@#/%=~_|]";
```
Comment (rickif) [inline fe/fe-core/src/main/java/com/starrocks/sql/ast/CreateRoutineLoadStmt.java:139]: How about using a stricter regular expression to match IPv6 address?

Comment (qingzhongli) [reply fe/fe-core/src/main/java/com/starrocks/sql/ast/CreateRoutineLoadStmt.java:139]: Can you provide an example? I think more strict may lead to more complexity.

Comment (kimmking) [reply fe/fe-core/src/main/java/com/starrocks/sql/ast/CreateRoutineLoadStmt.java:139]: Variable name "IPV6_REGEX" is better?

Comment (qingzhongli) [reply fe/fe-core/src/main/java/com/starrocks/sql/ast/CreateRoutineLoadStmt.java]: Exactly. I updated the current PR, please review again.

**Review by kimmking — APPROVED** [review]

**Review by rickif — APPROVED** [review]

**Comment by qingzhongli** [comment]
@Mergifyio rebase

**Comment by qingzhongli** [comment]
@Mergifyio rebase

**Comment by rickif** [comment]
@qingzhongli 
Please sign the CLA for all commits.
You could run `git rebase HEAD~8 --signoff` and submit again.

**Comment by qingzhongli** [comment]
@rickif @kimmking Thank you for your patient reviews. This PR has some mistakes, I will close it and submit a new PR.

**Pull request closed without merging**